Source/HTTPMethod.swift

```diff
@@ -25,23 +25,32 @@
 /// HTTP method definitions.
 ///
 /// See https://tools.ietf.org/html/rfc7231#section-4.3
-public enum HTTPMethod: String {
+public struct HTTPMethod: RawRepresentable, Equatable, Hashable {
     /// `CONNECT` method.
-    case connect = "CONNECT"
+    public static let connect = HTTPMethod(rawValue: "CONNECT")
     /// `DELETE` method.
-    case delete  = "DELETE"
+    public static let delete = HTTPMethod(rawValue: "DELETE")
     /// `GET` method.
-    case get     = "GET"
+    public static let get = HTTPMethod(rawValue: "GET")
     /// `HEAD` method.
-    case head    = "HEAD"
+    public static let head = HTTPMethod(rawValue: "HEAD")
     /// `OPTIONS` method.
-    case options = "OPTIONS"
+    public static let options = HTTPMethod(rawValue: "OPTIONS")
     /// `PATCH` method.
-    case patch   = "PATCH"
+    public static let patch = HTTPMethod(rawValue: "PATCH")
     /// `POST` method.
-    case post    = "POST"
+    public static let post = HTTPMethod(rawValue: "POST")
     /// `PUT` method.
-    case put     = "PUT"
+    public static let put = HTTPMethod(rawValue: "PUT")
     /// `TRACE` method.
-    case trace   = "TRACE"
+    public static let trace = HTTPMethod(rawValue: "TRACE")
+
+    public let rawValue: String
+    
+    /// Creates a new instance with the `uppercased()` version of the specified `rawValue`.
+    /// 
+    /// - Parameter rawValue: The raw `String` value, stored `uppercased`.
+    public init(rawValue: String) {
+        self.rawValue = rawValue.uppercased()
```

Comment:
Is there any reason @jshier that we'd not want to uppercase the `rawValue`?

Reply:
Only if we want to support case-sensitive methods (e.g. `get` vs. `GET` vs. `GeT`), which I don't think we do. I'll take another look at the standard though.

Reply:
Actually, it looks like [HTTP methods are considered case-sensitive](https://tools.ietf.org/html/rfc7230#section-3.1.1). I'm not sure how much that's used in practice though. To be save it does seem like I should remove the `uppercased()` and make a not about the sensitivity.

Reply:
Updated in https://github.com/Alamofire/Alamofire/pull/2901/commits/83732f1549244cdebbab05998ac86583ad5001d1.